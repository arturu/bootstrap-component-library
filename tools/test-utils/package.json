{
  "name": "@openeuropa/bcl-test-utils",
  "version": "0.1.3",
  "description": "A local test helper",
  "main": "index.js",
  "private": true,
  "devDependencies": {
    "deepmerge": "4.2.2"
  }
}
