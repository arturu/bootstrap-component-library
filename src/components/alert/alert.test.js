import { merge, renderTwigFileAsNode } from "@oe/test-utils";

describe("OE - Alert", () => {
  const template = "@oe/alert/alert.html.twig";
  const render = (params) => renderTwigFileAsNode(template, params);

  const variants = [
    "alert-primary",
    "alert-secondary",
    "alert-success",
    "alert-danger",
    "alert-warning",
    "alert-info",
    "alert-light",
    "alert-dark",
  ];

  variants.forEach((variant) => {
    test(`- ${variant} renders correctly`, () => {
      expect.assertions(1);

      return expect(render({ variant })).resolves.toMatchSnapshot();
    });
  });
});
