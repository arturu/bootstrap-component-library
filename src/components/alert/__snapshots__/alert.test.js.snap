// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OE - Alert - alert-danger renders correctly 1`] = `
<test>
  <div
    class="alert alert-danger"
    role="alert"
  >
    
  A simple primary alert—check it out!

  </div>
</test>
`;

exports[`OE - Alert - alert-dark renders correctly 1`] = `
<test>
  <div
    class="alert alert-dark"
    role="alert"
  >
    
  A simple primary alert—check it out!

  </div>
</test>
`;

exports[`OE - Alert - alert-info renders correctly 1`] = `
<test>
  <div
    class="alert alert-info"
    role="alert"
  >
    
  A simple primary alert—check it out!

  </div>
</test>
`;

exports[`OE - Alert - alert-light renders correctly 1`] = `
<test>
  <div
    class="alert alert-light"
    role="alert"
  >
    
  A simple primary alert—check it out!

  </div>
</test>
`;

exports[`OE - Alert - alert-primary renders correctly 1`] = `
<test>
  <div
    class="alert alert-primary"
    role="alert"
  >
    
  A simple primary alert—check it out!

  </div>
</test>
`;

exports[`OE - Alert - alert-secondary renders correctly 1`] = `
<test>
  <div
    class="alert alert-secondary"
    role="alert"
  >
    
  A simple primary alert—check it out!

  </div>
</test>
`;

exports[`OE - Alert - alert-success renders correctly 1`] = `
<test>
  <div
    class="alert alert-success"
    role="alert"
  >
    
  A simple primary alert—check it out!

  </div>
</test>
`;

exports[`OE - Alert - alert-warning renders correctly 1`] = `
<test>
  <div
    class="alert alert-warning"
    role="alert"
  >
    
  A simple primary alert—check it out!

  </div>
</test>
`;
